bin/ansible2gdeploy.py

```diff
@@ -64,33 +96,61 @@ def main():
 
     # get machines from the inventory file
     servers = inventory.options(GLUSTER_SERVER_GROUP)
-    if gdeploy_conf.has_section("clients"):
+    if any([gc.has_section("clients") for gc in gdeploy_confs.values()]):
         clients = inventory.options(GLUSTER_CLIENT_GROUP)
     else:
         clients = []
-    print("servers: " + ", ".join(servers), file=sys.stderr)
-    print("clients: " + ", ".join(clients), file=sys.stderr)
-
-    # ignore hosts section if present
-    if gdeploy_conf.has_section("hosts"):
-        gdeploy_conf.remove_section("hosts")
-    # add servers into hosts sections
-    gdeploy_conf.add_section("hosts")
-    for server in servers:
-        gdeploy_conf.set("hosts", server, None)
 
-    if gdeploy_conf.has_section("clients"):
-        gdeploy_conf.set("clients", "hosts", ",".join(clients))
+    # prepare list of storage devices
+    storage_devices = []
+    if args.storage_devices:
+        storage_devices = args.storage_devices.split(",")
 
-    if args.dry_run:
-        gdeploy_conf.write(sys.stdout)
-    else:
-        if args.output is not None:
-            output_filename = args.output
+    print("servers: " + ", ".join(servers), file=sys.stderr)
+    print("clients: " + ", ".join(clients), file=sys.stderr)
+    print("devices: " + ", ".join(storage_devices), file=sys.stderr)
+
+    # update gdeploy config files
+    for gdeploy_conf_file in gdeploy_confs:
+        # ignore hosts section if present
+        if gdeploy_confs[gdeploy_conf_file].has_section("hosts"):
+            gdeploy_confs[gdeploy_conf_file].remove_section("hosts")
+        # add servers into hosts sections
+        gdeploy_confs[gdeploy_conf_file].add_section("hosts")
+        for server in servers:
+            gdeploy_confs[gdeploy_conf_file].set("hosts", server, None)
+
+        # configure client
+        if gdeploy_confs[gdeploy_conf_file].has_section("clients"):
+            gdeploy_confs[gdeploy_conf_file].set("clients", "hosts", ",".join(clients))
+
+        # configure storage devices
+        if args.storage_devices and \
+                gdeploy_confs[gdeploy_conf_file].has_section("backend-setup") and \
+                gdeploy_confs[gdeploy_conf_file].has_option("backend-setup", "devices"):
+
+            # number of used devices
+            n_devices = len(gdeploy_confs[gdeploy_conf_file].get("backend-setup", "devices").split(","))
+
+            if n_devices > len(storage_devices):
+                msg = "Not enough storage devices for {} - available: {} ({}), required: {}".format(
+                    gdeploy_conf_file, len(storage_devices), ",".join(storage_devices), n_devices)
+                print(msg, file=sys.stderr)
+                return 1
+            gdeploy_confs[gdeploy_conf_file].set("backend-setup", "devices", \
+                ",".join(storage_devices[:n_devices]))
```

Comment:
I would add here some logging message (in the same way as we do for `print("servers: " ...`) about device allocation, something like:
```
gdeploy_config/volume_gama_disperse_4_plus_2x2.create.conf was assigned devices: sdc,sdd
```

so that when you run it with multiple gdeploy config files, and the 1st one uses them all, the error would be more clear than:

```
devices: sda, sdb, sdc, sdd, sdf
Not enough storage devices for gdeploy_config/volume_gama_disperse_4_plus_2x2.create.conf - available: 0 (), required: 2  
```

Reply:
good point, should be fixed by https://github.com/usmqe/usmqe-setup/pull/172/commits/45e77f7e8e517f19676f30f9be6747aee3721e65